tests/acceptance/features/bootstrap/Provisioning.php

```diff
@@ -3120,58 +3120,36 @@ public function userTriesToSendGroupCreationRequestUsingTheAPI(string $user, str
 	 */
 	public function createTheGroup(string $group, ?string $method = null):void {
 		//guess yourself
-		if ($method === null) {
-			if ($this->isTestingWithLdap()) {
-				$method = "ldap";
-			} elseif (OcisHelper::isTestingWithGraphApi()) {
-				$method = "graph";
-			} else {
-				$method = "api";
-			}
-		}
 		$group = \trim($group);
-		$method = \trim(\strtolower($method));
 		$groupCanBeDeleted = false;
 		$groupId = null;
-		switch ($method) {
-			case "api":
-				$result = UserHelper::createGroup(
-					$this->getBaseUrl(),
-					$group,
-					$this->getAdminUsername(),
-					$this->getAdminPassword(),
-					$this->getStepLineRef()
-				);
-				if ($result->getStatusCode() === 200) {
-					$groupCanBeDeleted = true;
-				} else {
-					throw new Exception(
-						"could not create group '$group'. "
-						. $result->getStatusCode() . " " . $result->getBody()
-					);
-				}
-				break;
-			case "ldap":
+		if ($method === null) {
+			if ($this->isTestingWithLdap()) {
 				try {
 					$this->createLdapGroup($group);
 				} catch (LdapException $e) {
 					throw new Exception(
 						"could not create group '$group'. Error: $e"
 					);
 				}
-				break;
-			case "graph":
-				$newGroup = $this->graphContext->adminHasCreatedGroupUsingTheGraphApi($group);
+			} elseif (OcisHelper::isTestingWithGraphApi()) {
+				$newGroup = $this->graphContext->createGroup($group);
+				if ($newGroup->getStatusCode() === 201) {
+					$newGroup = $this->getJsonDecodedResponse($newGroup);
+				}
 				$groupCanBeDeleted = true;
 				$groupId = $newGroup["id"];
-				break;
-			default:
+			} else {
 				throw new InvalidArgumentException(
 					"Invalid method to create group '$group'"
 				);
+			}
 		}
-
 		$this->addGroupToCreatedGroupsList($group, true, $groupCanBeDeleted, $groupId);
+		Assert::assertTrue(
+			$this->groupExists($group),
+			"Group '$group' should exist but does not exist"
+		);
```

Comment:
I think this assert should also be the part of Given/Then step rather than the core creation function.
```suggestion
	
```

Reply:
Good point!